src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -236,21 +237,30 @@ private BoundStatement CollectDynamicAnalysis(BoundStatement original, BoundStat
             return statementFactory.StatementList(AddAnalysisPoint(SyntaxForSpan(original), statementFactory), rewritten);
         }
 
-        private BoundStatement AddAnalysisPoint(CSharpSyntaxNode syntaxForSpan, SyntheticBoundNodeFactory statementFactory)
+        private Cci.DebugSourceDocument GetSourceDocument(CSharpSyntaxNode syntax)
         {
-            // Add an entry in the spans array.
+            return GetSourceDocument(syntax, syntax.GetLocation().GetMappedLineSpan());
+        }
 
-            Location location = syntaxForSpan.GetLocation();
-            FileLinePositionSpan spanPosition = location.GetMappedLineSpan();
-            string path = spanPosition.Path;
+        private Cci.DebugSourceDocument GetSourceDocument(CSharpSyntaxNode syntax, FileLinePositionSpan span)
+        {
+            string path = span.Path;
             // If the path for the syntax node is empty, try the path for the entire syntax tree.
             if (path.Length == 0)
             {
-                path = syntaxForSpan.SyntaxTree.FilePath;
+                path = syntax.SyntaxTree.FilePath;
             }
 
+            return _debugDocumentProvider.Invoke(path, "");
```

Comment:
I'll add a named argument. Named or unnamed, I don't know what the second argument means either.